impress-org/givewp-next-gen
# New: Introduce the Template API Framework

## Description

❗️ This requires https://github.com/impress-org/givewp/pull/6448 to work. ❗️

This PR introduces the Template API Framework. I say "framework" because it doesn't include a Template at this point, just the framework to support one. The goal of this framework is to make adding a Template _very easy_. In fact, ideally a template may only be some CSS that's applied to the default markup. That would be fine! But it may also overload specific nodes as is necessary.

I will say this is partial, but I wanted to get at least this much reviewed and merged. Here's what's remaining:
1. Make a way for templates to be registered in the back-end
2. Pass the template JS and CSS to the front-end
3. Provide a first-class way for nodes to be overloaded

This also introduces a number of the base nodes such as Name, Email, Amount, and so forth. I'm also passing the `useFormContext` to the window and using it in the `Amount` field as a proof-of-concept as to how the template and add-ons can interact with the form data.

Finally, I did some work to how the Form Builder schema is converted to the Field API.

## Visuals

<img width="684" alt="image" src="https://user-images.githubusercontent.com/2024145/173162107-14c2ad49-ebc3-4b2e-af10-1ae8d4d457a7.png">

## Testing Instructions

This is largely a code review, but you can also pull it all down and check out that everything works. Try adding some custom fields within the Form Builder as well!

## Pre-review Checklist
